src/TADA_DataRetrieval_mrb.R

```diff
@@ -161,245 +161,442 @@
     }
-    WQPquery <- c(WQPquery, endDate = endDate)
-  }
-  
-  # Retrieve all 3 profiles
-  print("Downloading WQP query results. This may take some time depending upon the query size.")
-  results.DR <- dataRetrieval::readWQPdata(WQPquery,
-                                           dataProfile = "resultPhysChem",
-                                           ignore_attributes = TRUE
-  )
-  # check if any results are available
-  if ((nrow(results.DR) > 0) == FALSE) {
-    print("Returning empty results dataframe: Your WQP query returned no results (no data available). Try a different query. Removing some of your query filters OR broadening your search area may help.")
-    TADAprofile.clean <- results.DR
-  } else {
-    sites.DR <- dataRetrieval::whatWQPsites(WQPquery)
     
-    projects.DR <- dataRetrieval::readWQPdata(WQPquery,
-                                              ignore_attributes = TRUE,
-                                              service = "Project"
-    )
+    if (length(project) > 1) {
+      WQPquery <- c(WQPquery, project = list(project))
+    } else if (!is.null(project)) {
+      WQPquery <- c(WQPquery, project = project)
+    }
     
-    TADAprofile <- TADA_JoinWQPProfiles(
-      FullPhysChem = results.DR,
-      Sites = sites.DR,
-      Projects = projects.DR
-    )
+    if (length(providers) > 1) {
+      WQPquery <- c(WQPquery, providers = list(providers))
+    } else if (!is.null(providers)) {
+      WQPquery <- c(WQPquery, providers = providers)
+    }
     
-    # need to specify this or throws error when trying to bind rows. Temporary fix for larger
-    # issue where data structure for all columns should be specified.
-    cols <- names(TADAprofile)
+    if (length(organization) > 1) {
+      WQPquery <- c(WQPquery, organization = list(organization))
+    } else if (!is.null(organization)) {
+      WQPquery <- c(WQPquery, organization = organization)
+    }
     
-    TADAprofile <- TADAprofile %>% dplyr::mutate_at(cols, as.character)
+    if (length(endDate) > 1) {
+      if (is.na(suppressWarnings(lubridate::parse_date_time(endDate[1], orders = "ymd")))) {
+        stop("Incorrect date format. Please use the format YYYY-MM-DD.")
+      }
+      WQPquery <- c(WQPquery, endDate = list(endDate))
+    } else if (!is.null(endDate)) {
+      if (is.na(suppressWarnings(lubridate::parse_date_time(endDate, orders = "ymd")))) {
+        stop("Incorrect date format. Please use the format YYYY-MM-DD.")
+      }
+      WQPquery <- c(WQPquery, endDate = endDate)
+    }
     
-    # run TADA_AutoClean function
-    if (applyautoclean == TRUE) {
-      print("Data successfully downloaded. Running TADA_AutoClean function.")
-      
-      TADAprofile.clean <- TADA_AutoClean(TADAprofile)
+    # Retrieve all 3 profiles
+    print("Downloading WQP query results. This may take some time depending upon the query size.")
+    print(WQPquery)
+    results.DR <- dataRetrieval::readWQPdata(WQPquery,
```

Comment:
would swapping the order in which results.DR and sites.DR are created speed things up? Thinking about how we could split up the `readWQPdata()` in a similar fashion to the {sf} pull if we do it that way....